### DISCUSSION TITLE
Support for Descript.ion

### DISCUSSION THREAD

**Discussion by ibulis1864** [sumatrapdfreader/sumatrapdf#3866] [Ideas]
I have lots of PDF files and just skimmed them through. I could make notes in the PDF files but it's tough for searching outside.
So I wish to add a comment  for the whole PDF file and saving that in the Descript.ion file so that I can get a brief description when cursor hovers over them.
It's easy to add description in a file mannager like Totalcommander But it will be more convenient.

**Comment by GitHubRulesOK** [comment]
Descript.ion is a throw back to the days of 4DOS or Dos4 I remember it as if it was just a century ago
 however its nowhere near as good as find word in windows explorer or any paid up indexing service. How on earth would SumatraPDF be able to do any different to your own personal use of NotePad in a folder? Many files have no MetaData to use and you can easily run command line tools to extract 100% of a files text without using a viewer

If you have a page of text and want to copy paste a paragraph simply open Notepad alongside SumatraPDF that can even send the current filename to a windows batch file for your custom notes.

hmm so long ago since somebody asked for something similar but see this method https://forum.sumatrapdfreader.org/t/advanced-tips-add-your-own-notes/2688

**Reply by ibulis1864** [reply]
I copy the ExternalViewers content but it just open the current pdf file with notepad. So I checked that and found it always try to open pdffilename.pdf".Page1.txt. These's always  one more quote and I dont know why. notepad will ignore the argument after the quote .I don't know the bug come from Windows or command-line or Sumatra but it just doesn't work.

**Reply by ibulis1864** [reply]
I checked other special values %p and %d and it works well. So I think it may be a bug.

**Reply by GitHubRulesOK** [reply]
That example was for page by page notes in earlier versions, it has a few different needed change for delays based on system speed.

Just offered as a template for your adaptation you can reuse it per the comments to not be page by page, 
intended to be trimmed down, and you can develop a better system especially for your custom use.

I dont normally use the script but, here as an example I am using Notepad with one file per file , but if using daily I would use write editor as more powerful and can paste image as well as text into a docx.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/693e48d5-e311-4d77-988f-6250d1283fd1)

**Comment by GitHubRulesOK** [comment]
@ibulis1864 

Yes there is a bug in how recent releases  handle those commands 
an updated method was added yesterday in pre-release 3.6.15888 so see changes in https://github.com/sumatrapdfreader/sumatrapdf/issues/3868

for a specific choice of command please list what you are trying but for now try **3.6.15888** with
```
ExternalViewers [
	[
		CommandLine = "c:\windows\System32\notepad.exe" """%1-page%p.txt"""
		Name = Notepad Page &Txt
		Filter = *.*
	]
]
```
English shortcut would be `Alt F T`
